src/os/bluestore/Compression.cc

```diff
@@ -162,7 +162,7 @@ int32_t Estimator::split_and_compress(
   uint32_t size = data_bl.length();
   ceph_assert(size > 0);
   uint32_t blobs = (size + max_blob_size - 1) / max_blob_size;
-  uint32_t blob_size = p2roundup(size / blobs, au_size);
+  uint32_t blob_size = p2roundup((size + blobs - 1) / blobs, au_size);
   std::vector<uint32_t> blob_sizes(blobs);
   for (auto& i: blob_sizes) {
     i = std::min(size, blob_size);
```

Comment:
the std::min() should probably be moved out of the loop?

Reply:
A matter of taste, but I would have preferred a more descriptive version of lines 164,165.
Possibly something along the lines of:
double blobs_as_double = std::ceil(static_cast<double>(size) / max_blob_size); 
uint32_t blobs = blobs_as_double;
uint32_t blob_size = p2roundup(std::ceil( size / blobs_as_double), au_size);

Reply:
`std::min()` cannot be outside loop.
It assigns values to subsequent elements, so the sum of all elements will be size.
But you missing it suggest to me that maybe I should rename "i" to "iterated_blob_size"?

`p2roundup()` cannot be used, blobs is not p2 value.
What I would like to do, but it is not if it exist would be: `std::divide_roundup()`.
I would prefer to stay in integer arithmetic and do not use floats here.

Reply:
- yes. I missed that. Shame on me.
- p2roundup() implementation does not require p2, it seems. But you are right - even if so, it's not guaranteed.
- I understand the reluctance to move to floats. Was looking for such a divide_roundup(), but did not find one yet.